Source/Endpoint.swift

```diff
@@ -74,16 +75,25 @@ extension Endpoint {
 
         return parameterEncoding.toAlamofire.encode(request, parameters: parameters).0
     }
+
+    // let's return a mutable request, that could be modify before it's be performed
+    func toNSMutableURLRequest() -> NSMutableURLRequest {
```

Comment:
I think we should probably just get Alamofire to create the `NSURLRequest`, then call `mutableCopy` on it. Does that make sense?